client/src/components/pages/EditSurvey.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+
+import { Page, Spinner, Stack } from "@shopify/polaris";
+import { EditSurveyForm } from "..";
+
+export function EditSurvey() {
+  let { id: surveyId } = useParams();
+
+  const [name, setName] = useState("");
+  const [questions, setQuestions] = useState([]);
+
+  // a curry function that we will pass down to EditSurveyForm to update state
+  // This will be our state updated for all our survey files
+  // index will be the index of the question in in survey.questions,
+  // extraOpts is for updating mins/max in range questions or options in option questions
+  function handleQuestionChange(index, extraOpts) {
```

Comment:
@kirinrastogi , The [QuestionList](https://github.com/vikrambombhi/SimpleSurvey/blob/master/client/src/components/QuestionList.js) state management was a good initial stab but it becomes problematic when sharing with other components.

If you're curious about a better pattern to use, this `EditSurvey` and `EditSurveyForm` handles state management quite well. `EditSurvey` is responsible for owning and updating the state. `EditSurveyForm` is a _dumb_ component only responsible for rendering what it's given and updates state with the handlers passed to it. In this case, it's this `handleQuestionChange` function